crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -1196,22 +1194,10 @@ where
                             ));
                         }
                     } else if alias.node.name == "*" {
-                        self.add_binding(
-                            "*",
-                            Binding {
-                                kind: BindingKind::StarImportation(StarImportation {
-                                    level: *level,
-                                    module: module.clone(),
-                                }),
-                                runtime_usage: None,
-                                synthetic_usage: None,
-                                typing_usage: None,
-                                range: Range::from(stmt),
-                                source: Some(*self.ctx.current_stmt()),
-                                context: self.ctx.execution_context(),
-                                exceptions: self.ctx.exceptions(),
-                            },
-                        );
+                        self.ctx.scope_mut().star_imports.push(StarImportation {
```

Comment:
Nit: It would be a good practice to start, in my view not exposing the internal data structures directly and instead, add public methods on `Scope` to add imports and read them.

Doing so eases replacing the internal representation later or uphold constraints that must be guaranteed (e.g. inserting in sorted order)

Reply:
Fixed :)